### PR TITLE
New rule: Link name (alone) is descriptive

### PR DESCRIPTION
**Pull request by Jym77**
New rule for 2.4.9, based on the existing one for 2.4.6.

Closes issue(s):

- closes N/A

Need for Final Call:
This will require a 2 weeks Final Call (new rule)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [ ] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [ ] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/link-alone-descriptive-aizyf1.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+```html
+<p>See the description of <a href="#desc">this product</a>.</p>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 5
+
+These links have [accessible name][] which, alone, do not describe their purpose.
```
Comment (WilcoFiers) [inline _rules/link-alone-descriptive-aizyf1.md:168]:
```suggestion
These links have [accessible names][accessible name] which, alone, do not describe their purpose.
```

File: _rules/link-alone-descriptive-aizyf1.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+This link is not [included in the accessibility tree][].
+
+```html
+<a href="https://www.w3.org/WAI" style="display: none;"
+	><img src="/test-assets/5effbb/cart.svg" alt="Checkout" />Checkout</a
+>
+```
+
+#### Inapplicable Example 3
```
Comment (WilcoFiers) [inline _rules/link-alone-descriptive-aizyf1.md:207]: Don't think we need this TBH. If you can inapplicable 1 and 2 right, you won't get this wrong.

File: _rules/link-in-context-descriptive-5effbb.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ id: 5effbb
 name: Link in context is descriptive
 rule_type: atomic
 description: |
-  This rule checks that the accessible name of a link together with its context describe its purpose.
+  This rule checks that the accessible name of a link together with its context describes its purpose.
```
Comment (WilcoFiers) [inline _rules/link-in-context-descriptive-5effbb.md:6]: I think these changes should go into a separate PR.

File: pages/glossary/semantic-link.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+title: Semantic link
+key: semantic-link
+unambiguous: true
+objective: true
+input_aspects:
+  - Accessibility tree
+  - DOM tree
+---
+
+A semantic link is any HTML element which has the [semantic role][] of [`link`][link] or a [semantic role][] that inherits from the [`link`][link] role.
```
Comment (WilcoFiers) [inline pages/glossary/semantic-link.md:11]: Why only an HTML element? If we make this HTML and SVG, we can remove the use of that phrase from the applicability.

**Review by WilcoFiers — DISMISSED** [review]

File: _rules/link-in-context-descriptive-5effbb.md (hunk 3 above)
Comment (Jym77) [reply _rules/link-in-context-descriptive-5effbb.md:6]: Good point. I'm grouping all changes in the existing rule in #1530 (including changes requested here that make sense to be backported).

File: pages/glossary/semantic-link.md (hunk 4 above)
Comment (Jym77) [reply pages/glossary/semantic-link.md:11]: Good point. I actually think the definition (and the rule) can be made technology agnostic since the concept s involved (role, name) are.

File: _rules/link-alone-descriptive-aizyf1.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+- This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content), and sufficiently understood.
+
+## Accessibility Support
+
+- Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `link` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+
+## Background
+
+- Related rule:
+  - [Link has non-empty accessible name](https://act-rules.github.io/rules/c487ae)
```
[the remark this answers is not in the document]
Comment (daniel-montalvo) [reply _rules/link-alone-descriptive-aizyf1.md:47]: +1 to leave as is and discuss this more broadly (maybe in the ACT-R CG context?)

[the remark this answers is not in the document]
Comment (Jym77) [reply _rules/link-alone-descriptive-aizyf1.md:47]: Issue #1531

**Review by carlosapaduarte — APPROVED** [review]

File: _rules/link-alone-descriptive-aizyf1.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+- [ARIA7: Using aria-labelledby for link purpose](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA7)
+- [ARIA8: Using aria-label for link purpose](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This link has an [accessible name][] (from its text) describing its purpose.
```
Comment (daniel-montalvo) [inline _rules/link-alone-descriptive-aizyf1.md:61]:
```suggestion
This `a` element has an [accessible name][] from its text that describes its purpose.
```

File: _rules/link-alone-descriptive-aizyf1.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+```html
+<a href="#desc">See the description of this product.</a>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 2
+
+This link has an [accessible name][] (from the `alt` attribute on the image) describing its purpose.
```
Comment (daniel-montalvo) [inline _rules/link-alone-descriptive-aizyf1.md:71]:
```suggestion
This `a element has an [accessible name][] from the `alt` attribute on its `img` element that describes its purpose.
```

File: _rules/link-alone-descriptive-aizyf1.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+<a href="#main"><img src="/test-assets/5effbb/main.png" alt="Go to the main content"/></a>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Passed Example 3
+
+This link has an [accessible name][] (from its text) describing its purpose.
```
Comment (daniel-montalvo) [inline _rules/link-alone-descriptive-aizyf1.md:83]:
```suggestion
This `span` element has an [explicit role][] of `link` and an [accessible name][] from its text that describes its purpose.
```

File: _rules/link-alone-descriptive-aizyf1.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+```html
+<span role="link" tabindex="0" onclick="document.location+='#desc'">See description of the product.</span>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Passed Example 4
+
+This link has an [accessible name][] (from its `aria-labelledby` attribute) describing its purpose.
```
Comment (daniel-montalvo) [inline _rules/link-alone-descriptive-aizyf1.md:93]:
```suggestion
This `a` element has an [accessible name][] from its `aria-labelledby` attribute that describes its purpose.
```

**Review by daniel-montalvo — APPROVED** [review]
I think these suggestions better align with the descriptions of the related rule. Will be providing suggestions down below for the other test cases.

I would suggest to take this approach for the remaining test case descriptions.

Minor issue anyway, I leave to editor's discretion.

File: _rules/link-alone-descriptive-aizyf1.md (hunk 2 above)
Comment (WilcoFiers) [reply _rules/link-alone-descriptive-aizyf1.md:206]: Hmm, ok, not a deal breaker. Feel free to leave it in.

**Review by WilcoFiers — DISMISSED** [review]
Couple editorials

**Review by WilcoFiers — APPROVED** [review]

**Comment by Jym77** [comment]
Final Call ends on February 1st.

**Comment by Jym77** [comment]
Final Call has ended. Merging.
